### PR TITLE
Added Multiple basic ufuncs

### PR DESCRIPTION
**Pull request by ShahzadUmair**
The ufuncs this PR covers include:

- add
- subtract
- multiply
- matmul
- divide
- negative
- positive
- power
- fmod
- square
- greater
- logaddexp

As of now these ufuncs only supports 1D (float and double) Views as input .

### CODE REVIEW

**Review by NaderAlAwar — CHANGES_REQUESTED** [review]
Looks good overall, and I think it should be ready to merge after the changes. Let's see if Tyler has any comments.

**Review by tylerjereddy — COMMENTED** [review]
I tried to add some suggestions. Two other things to keep an eye on:

- updating gh-62 as we move forward
- test suite runtime jumps from about 3:30 to 5:30 on my machine with the extra testing here; I don't have concrete suggestions for that at this time, but maybe some efficiency improvements + gh-61 type stuff (tracking which types we support for a given ufunc may also get annoying)

File: pykokkos/lib/ufuncs.py
Comment (tylerjereddy) [inline pykokkos/lib/ufuncs.py:403]: this is the naming thing I was talking about--`viewB` is clearly two dimensional here--maybe folks with a linalg background will have a better kernel name suggestion than `..1d..`, but may not matter for now. Or I guess you could argue it is a "column vector" that is conceptually 1D, dunno.

File: pykokkos/lib/ufuncs.py
Comment (ShahzadUmair) [reply pykokkos/lib/ufuncs.py]: Removed the Range Policy altogether and replaced with `view.shape[0]`. Please let me know if this is the preferred way.

File: pykokkos/lib/ufuncs.py
Comment (ShahzadUmair) [reply pykokkos/lib/ufuncs.py]: So I followed the numpy's implementation for ufuncs and they always return the result in a new array. To make sure we use the same name everywhere, I renamed the output view in other funcs to "out" as well (it was previously called "view"). 

I also removed the `out = None` because it isnt needed.

File: tests/test_ufuncs.py
Comment (ShahzadUmair) [reply tests/test_ufuncs.py]: I missed reverting this back to `numpy_dtype` probably. Reverted.

File: pykokkos/lib/ufuncs.py
Comment (NaderAlAwar) [reply pykokkos/lib/ufuncs.py]: @tylerjereddy by default numpy allocates a new array  for all ufuncs right? But that isn't the approach we've been taking right? We've just been modifying the array in place?

File: pykokkos/lib/ufuncs.py
Comment (NaderAlAwar) [reply pykokkos/lib/ufuncs.py:301]: Yeah we will need to add broadcasting eventually

File: pykokkos/lib/ufuncs.py
Comment (NaderAlAwar) [reply pykokkos/lib/ufuncs.py]: I think with the change this is fine for now, but eventually we will need a better system for checking view types

File: pykokkos/lib/ufuncs.py
Comment (NaderAlAwar) [reply pykokkos/lib/ufuncs.py:439]: Agreed, I'll open new issues for checking shapes and types

File: pykokkos/lib/ufuncs.py
Comment (NaderAlAwar) [reply pykokkos/lib/ufuncs.py:422]: I think we should add this now for completeness' sake, but eventually we should compare our implementations to numpy and optiminze them

File: pykokkos/lib/ufuncs.py
Comment (tylerjereddy) [reply pykokkos/lib/ufuncs.py]: For `unary` ufuncs I think we've typically been modifying in place, yes. But most of the ufuncs here are `binary`, with two view operands, so it is unclear how you'd do anything other than provide a new view with the output. An `out` argument could be added later to control the location of the output result a bit more, but not sure we'd want to prioritize that just yet?

I certainly don't mind if we start providing a new view with the results for the `unary` cases as well--that's probably closer to what NumPy does, which in turn probably makes it easier for us to show performance improvements without changing the meaning in commonly-use algos.

File: pykokkos/lib/ufuncs.py
Comment (NaderAlAwar) [reply pykokkos/lib/ufuncs.py]: I think we should use a smaller datatype, but did we ever decide if we will be adding the `bool` datatype to `pykokkos-base`?

Comment (tylerjereddy) [reply pykokkos/lib/ufuncs.py]: > did we ever decide if we will be adding the bool datatype to pykokkos-base

I don't think we reached a final decision, but we probably should for array API conformance, rather than the current `bool` hacks we have over here. I think the NumPy `bool` type is a full byte per i.e., https://stackoverflow.com/questions/5602155/numpy-boolean-array-with-1-bit-entries.

Would we have a compelling reason to try to get away with less than 8 bits for our `bool`? Maybe we need an issue/discussion where we look at what NumPy vs. C++ do with `bool` arrays, and how hard it would be to get `repr` that shows `True` and `False` instead of `1` and `0` when you print, etc.

File: tests/test_ufuncs.py
Comment (NaderAlAwar) [reply tests/test_ufuncs.py:379]: @tylerjereddy so you think these names are fine?

File: pykokkos/lib/ufuncs.py
Comment (NaderAlAwar) [reply pykokkos/lib/ufuncs.py]: Yeah I think we do need an `out` argument eventually. I think that right now for the binary ufuncs it's fine to just allocate and return a new array.

File: pykokkos/lib/ufuncs.py
Comment (NaderAlAwar) [reply pykokkos/lib/ufuncs.py]: Yeah we should definitely add `bool` when we get the chance. It would be nice to have less than 8 bits for `bool` but I don't think that's high priority for now.

File: pykokkos/lib/ufuncs.py
Comment (NaderAlAwar) [reply pykokkos/lib/ufuncs.py:418]: I think these names are fine for now but matmul will definitely need to be revisited in the future for both this issue and performance

File: tests/test_ufuncs.py
Comment (tylerjereddy) [reply tests/test_ufuncs.py:379]: I don't have a better idea at this time anyway

**Pull request merged**
